tidb_exporter/main.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"database/sql"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ngaut/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+// Exporter collects database query stats and exports them using
+// the prometheus metrics package.
+type Exporter struct {
+	db       *sql.DB
+	tidbOpts tidbOpts
+}
+
+type tidbOpts struct {
+	address  string
+	username string
+	password string
+}
+
+// NewExporter returns an initialized Exporter.
+func NewExporter(opts tidbOpts) (*Exporter, error) {
+	db, err := accessDatabase(opts.username, opts.password, opts.address, dbname)
+	if err != nil {
+		log.Errorf("access database '%s' error, %v", opts.address, err)
```

Comment:
u output it outside, dont need to output this error log again.   or u can use `juju/errors`'s `Annotationf` and `ErrorStack`